singularity-eos/eos/singularity_eos.cpp

```diff
@@ -208,9 +208,21 @@ int init_sg_eospac(const int matindex, EOS *eos, const int id, int const *const
   eos[matindex] = eos_.GetOnDevice();
   return 0;
 }
+
 int init_sg_eospac(const int matindex, EOS *eos, const int id) {
   return init_sg_eospac(matindex, eos, id, def_en, def_v);
 }
+
+// CMM: A more robust solution may be to preprocess the fortran,
+//      however that is slightly more involved than you'd like to think.
+#else // stubs if not built, just to satisfy linking.
+int init_sg_eospac(const int matindex, EOS *eos, const int id, int const *const enabled,
+                   double *const vals) {
+  return 0;
+}
+
+int init_sg_eospac(const int matindex, EOS *eos, const int id) { return 0; }
+
```

Comment:
this seems fine for now

Reply:
This may need some amending for GPU builds. I'm not sure the Darwin CI will pass. Lets get that passing and then merge in. But I agree that eventually we should use the preprocessor to block out unneeded functions. @mauneyc-LANL could you make an issue so that I don't forget to modify the fortran interface to use the preprocessor in the future?

Reply:
Just to make a note of it: this would apply to other "dependent" routines that are conditionally compiled (e.g. stuff with `SINGULARITY_USE_HDF5/SPINER_USE_HDF5`). For the moment I'm not aware of host-codes dependent on `HDF5` that need a Fortran interface, but likely will encounter a similar issue to the above if that every happens.

Reply:
> This may need some amending for GPU builds. I'm not sure the Darwin CI will pass. Lets get that passing and then merge in. But I agree that eventually we should use the preprocessor to block out unneeded functions. @mauneyc-LANL could you make an issue so that I don't forget to modify the fortran interface to use the preprocessor in the future?

I should probably make it clear here: the reason I didn't use preprocessor guards in the Fortran source is because this compilers all handle fortran preprocessing in slightly different ways (and AFAIK none do it by default), and this can lead to frustrations down the road (especially with CMake, with it's somewhat tepid Fortran support). I would argue against putting preprocessor directives into the Fortran source, if we can help it.

As for tests, this would only effect a build with `SINGULARITY_USE_FORTRAN=ON` & `SINGULARITY_USE_EOSPAC=OFF`, and as best as I can tell the only test where that is an issue would be the test I added with this PR.

Reply:
Why would this effect GPU builds at all @dholladay00 ?

Reply:
Because it will by default use fortran as the linker language and resolving relocatable device code at link time requires extra steps that the fortran linker won't know about unless to you tell cmake to tell the fortran linker to worry about it(it will require enabling cuda as one of the cmake project languages and then a special command for the exec). I now know how to make this work so it's not a big deal, but I'd be very surprised if this test compiles and links for GPU builds.

Reply:
@mauneyc-LANL can you give this a try?

Reply:
@Yurlungur @mauneyc-LANL we no longer need rdc

Reply:
:+1: